===== Patch instructions generated by SAVER =====
[+] { Insert: if ({ PortObjectAppend:n$11 == { -1 } }) free(*(PortObjectAppend:poia)) at 6 (line 1469, column 9) }
===== Actual patch in diff format =====
--- a/table1/benchmarks/snort-2.9.13/src/sfutil/sfportobject.c
+++ b/table1/benchmarks/snort-2.9.13/src/sfutil/sfportobject.c
@@ -1475,7 +1475,8 @@ PortObject * PortObjectAppend(PortObject * poa, PortObject * pob )
 
        memcpy(poia,poib,sizeof(PortObjectItem));
 
-       sflist_add_tail(poa->item_list,poia);
+       if (sflist_add_tail(poa->item_list,poia) == -1)
+          free(poia);
    }
    return poa;
 }
